### PR TITLE
Create Protected branch plugin

### PR DESCRIPTION
**Pull request by jBouyoud**
# What Changed

New plugin to handle protected branch that enforce administrator enforcements.

## Why

Todo: N/A

- [x] Add tests
- [x] Add docs

## Change Type

Indicate the type of change your pull request is:

- [ ]  `documentation`
- [ ]  `patch`
- [x]  `minor`
- [ ]  `major`

### CODE REVIEW

**Comment by jBouyoud** [comment]
@hipstersmoothie anything that I can do to help merge this PR ?

**Comment by Sy1v4in** [comment]
Hello 👋
Any update on this PR? I've tried it on my project and it works perfectly 👌

File: plugins/protected-branch/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      if (!protectedBranchPlugin?.[1]?.reviewerToken) {
+        auto.checkEnv(this.name, "PROTECTED_BRANCH_REVIEWER_TOKEN");
+      }
+    });
+
+    auto.hooks.publish.tapPromise(
+      {
+        name: this.name,
+        // Include this plugin in a high priority stage in order to be mostly often before others plugins
+        stage: -1,
```
Comment (hipstersmoothie) [inline plugins/protected-branch/src/index.ts:60]: I didn't even know this was a thing :O

File: plugins/protected-branch/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        auto.logger.log.info("Open a release PR to handle changes 🐙 ");
+        const prNumber = await gitOperator.createPr(
+          "Automatic release",
+          headBranch,
+          auto.baseBranch
+        );
+
+        // As reviewer, allowed in : `Restrict who can push to matching branches`
+        auto.logger.log.info("Mark released pr as reviewed ✅ ");
+        await gitOperator.approvePr(this.options.reviewerToken, prNumber, sha);
```
Comment (hipstersmoothie) [inline plugins/protected-branch/src/index.ts:96]: Wow that's clever. This is an impressive solution

**Review by hipstersmoothie — APPROVED** [review]
👏 Amazing job (sorry on the slow merge). This is a wonderful contribution..

**Comment by hipstersmoothie** [comment]
@jBouyoud Could you update the lock file?

**Comment by hipstersmoothie** [comment]
:rocket: PR was released in [`v10.38.0`](https://github.com/intuit/auto/releases/tag/v10.38.0) :rocket:

**Comment by jBouyoud** [comment]
🎉  Thanks

**Comment by jBouyoud** [comment]
@hipstersmoothie I just realize that I've an improvement on my side that I not push here (do not rely on gh CLI and use auti Git class instead in order to remove a dependency and handle rate limits). I'll try to open a new PR for that. 

I see that you have implemented a cleanup (to delete the branch) ; You also have to think that the branch could not exists anymore for people using Github built-in branch deletion after merge.

**Comment by hipstersmoothie** [comment]
Pr?

**Comment by jBouyoud** [comment]
Please find the mentioned PR [here](https://github.com/intuit/auto/pull/2317)

**Pull request merged**